dll/win32/shell32/CDefView.cpp

```diff
@@ -2781,45 +2781,42 @@ LRESULT CDefView::OnNotify(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &bHandl
             break;
         case LVN_BEGINLABELEDITW:
         {
-            DWORD dwAttr = SFGAO_CANRENAME;
+            TRACE("-- LVN_BEGINLABELEDITW %p\n", this);
+            HWND hEdit = ListView_GetEditControl(m_ListView);
             pidl = _PidlByItem(lpdi->item);
+            DWORD fAttr = pidl ? GetItemAttributes(pidl, SFGAO_CANRENAME | SFGAO_FOLDER | SFGAO_FILESYSTEM) : 0;
+            if (!hEdit || !(fAttr & SFGAO_CANRENAME))
+            {
+                MessageBeep(0xffffffff);
+                return TRUE;
+            }
 
-            TRACE("-- LVN_BEGINLABELEDITW %p\n", this);
+            WCHAR szName[MAX_PATH], *pszText = lpdi->item.pszText;
+            if (SUCCEEDED(Shell_DisplayNameOf(m_pSFParent, pidl, SHGDN_FOREDITING | SHGDN_INFOLDER,
+                                              szName, _countof(szName))))
+                ::SetWindowText(hEdit, pszText = szName);
```

Comment:
```suggestion
            if (SUCCEEDED(Shell_DisplayNameOf(m_pSFParent, pidl, SHGDN_FOREDITING | SHGDN_INFOLDER,
                                              szName, _countof(szName))))
            {
                pszText = szName;
                ::SetWindowText(hEdit, pszText);
            }
```
Please avoid these assignations in the middile of a function call.